shared/health/src/main/java/io/pravega/shared/health/Registry.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import java.util.Optional;
+
+public interface Registry<T> {
+    /**
+     * Register some object of arbitrary type to the registry.
```

Comment:
It's not an arbitrary type. It's of type `T`. Update this interface Javadoc as well.